### PR TITLE
fix(): change waitlist conversation creation

### PR DESCRIPTION
**Pull request by jcwirko**

### CODE REVIEW

File: src/cloud/validateCode.js [hunk 1]
```diff
@@ -60,52 +59,58 @@ const setUserStatus = async (user, reservation = null) => {
...
-      //   'invitation'
-      // );
-
-      const onboardingRole = await new Parse.Query(Parse.Role)
-        .equalTo('name', ONBOARDING_ADMIN)
-        .first();
-      if (onboardingRole) {
-        // If the role is defined, get the first user with it
+      const hasWaitListConversation = await ChatService.getConversationByCid(
+        `messaging:${user.id}_waitlist_conversation`,
```
Comment (jcwirko) [inline src/cloud/validateCode.js:63]: here we check if has some wailist conversation

File: src/cloud/validateCode.js [hunk 2]
```diff
@@ -60,52 +59,58 @@ const setUserStatus = async (user, reservation = null) => {
...
+
+      if (!hasWaitListConversation) {
+        // TODO: Uncomment when we use again the currentQuePosition logic.
+        // user.set('quePosition', currentQuePosition);
+        // if (maxQuePosition >= currentQuePosition) {
+        //   user.set('status', 'inactive');
+        // } else {
+        //   user.set('status', 'waitlist');
+        // }
+        user.set('status', 'waitlist');
```
Comment (Benjamindodgson) [inline src/cloud/validateCode.js:74]: This probably needs to get set before the if check. They should have a status of waitlist regardless if they have conversation or not.

File: src/cloud/validateCode.js [hunk 3]
```diff
@@ -60,52 +59,58 @@ const setUserStatus = async (user, reservation = null) => {
...
+        // TODO: Uncomment when the app (frontend) is ready to use it.
+        // await ChatService.createConversation(
+        //   user,
+        //   `${user.id}_invitation_conversation_${new Date().getTime()}`,
+        //   'invitation'
+        // );
+
+        const onboardingRole = await new Parse.Query(Parse.Role)
+          .equalTo('name', ONBOARDING_ADMIN)
```
Comment (Benjamindodgson) [inline src/cloud/validateCode.js:83]: This needs to use my phone number, 12068509234, not the onboarding admin.

File: src/cloud/validateCode.js [hunk 4]
```diff
@@ -164,17 +169,25 @@ const validateCode = async request => {
...
         );
 
-        const fromUser = await new Parse.Query(Parse.User).get(conversation.data.created_by.id);
+        const fromUser = await new Parse.Query(Parse.User).get(
+          conversation.data.created_by.id,
+        );
         const fullName = UserUtils.getFullName(fromUser);
-        const connection = await new Parse.Query('Connection').equalTo('channelSId', conversationCid).find({ useMasterKey: true });
+        const connection = await new Parse.Query('Connection')
+          .equalTo('channelSId', conversationCid)
```
Comment (Benjamindodgson) [inline src/cloud/validateCode.js:186]: ChannelCid not ChannelSid

**Review by Benjamindodgson — CHANGES_REQUESTED** [review]
See comments.

File: src/cloud/validateCode.js (hunk 4 above)
Comment (Benjamindodgson) [reply src/cloud/validateCode.js:186]: sorry this is initialConversations, which is an array of strings

Comment (Benjamindodgson) [reply src/cloud/validateCode.js:186]: conversationSid is no longer used.

Comment (jcwirko) [reply src/cloud/validateCode.js:186]: This code is for reservationId not for waitlist

File: src/cloud/validateCode.js [hunk 5]
```diff
@@ -60,52 +59,60 @@ const setUserStatus = async (user, reservation = null) => {
...
+        //   'invitation'
+        // );
+
+        // Retrieve the user with the phoneNumber
+        const userQuery = new Parse.Query(Parse.User);
+        userQuery.equalTo('phoneNumber', BENJI_PHONE_NUMBER);
+
+        const admin = await userQuery.first({ useMasterKey: true });
+
+        await ChatService.createConversation(
```
Comment (Benjamindodgson) [inline src/cloud/validateCode.js:90]: Add a check to see if a waitlist conversation has already been created for this user. They could login, delete app, come back in and still be on the waitlist, and then would have 2 waitlist conversations

**Pull request merged**